src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -149,6 +201,8 @@ public void removeTaskId(String taskId) {
         if (entry.getValue() != null && entry.getValue().equals(taskId)) {
           journalNodes.put(entry.getKey(), null);
           setJournalNodes(journalNodes);
+          Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+          deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
Bug? Seems weird to have a single timestamp for multiple JNs, or a single timestamp for multiple DNs (or NNs). They each die/recover independently, so I would think that the timestamp should be per node/host, and then you can remove a deadXNode that failed a while ago, but still wait for a more recently failed XNode to recover. Otherwise, if you keep getting a new DN failure every minute, and only half of them recover, you won't ever remove any of the first failures.